### ISSUE TITLE
Extension Disabled on Chrome Canary

### ISSUE THREAD

**Issue by maxvonhippel** [EFForg/privacybadger#1186]
I just opened Chrome Canary for the first time in a couple days to receive [this message](https://support.google.com/chrome/answer/2811969?visit_id=1-636218619297665908-2390945262&p=ui_remove_non_cws_extensions&rd=1).  The enable button is now greyed out on my extensions list, so I can't enable it.  

<img width="804" alt="badger" src="https://cloud.githubusercontent.com/assets/4683443/22623455/f26162d0-eb18-11e6-8baa-972ab4740e16.png">

Note that I get the same behavior in both Developer mode and non-Developer mode.  My Chrome shows a checkmark saying it is up-to-date.  I'm on Version 58.0.3002.0 canary (64-bit).

**Comment by ghostwords** [comment]
Something like this happened before, at least once: #916.

**Comment by maxvonhippel** [comment]
Sounds to me like there are [various hacks](http://superuser.com/questions/767286/re-enable-extensions-not-coming-from-chrome-web-store-on-chrome-v35-with-enhan) to get around this issue.  That said, I feel like it would be worth agreeing on some canonical method, and including instructions to that effect in the docs.  I'm happy to write the instructions and make a PR if someone with more experience in this project can say which direction we should go, but I don't want to presume to know what method of circumventing this rule would be the most appropriate to recommend here, since I haven't been involved in this project for very long.  Any thoughts/recommendations?

**Comment by cowlicks** [comment]
related to #1204

**Comment by maxvonhippel** [comment]
Thanks @cowlicks, looks like this happens [relatively](https://github.com/EFForg/privacybadger/issues/1204) [often](https://github.com/EFForg/privacybadger/issues/911).  Do you think I'm right that we should have a canonical set of instructions for dealing with this, since in all likelihood it will, even once resolved this time, occur again in the future?  In other words, I propose that the docs should say "in the event that you get error X, comment on issue Y on Github instead of making a new identical one as fools such as Max (me) have, and then use instructions X1...XN to temporarily avoid this issue so you can continue your development efforts" (or something a bit less snarky and along those lines).

**Comment by cowlicks** [comment]
@maxvonhippel that is a great idea. I don't think anyone reads the stuff in `docs/`, so we should probably include a link to you instructions in the README.

I'll poke the chrome dev team and see what the status of the fix for this is.

**Comment by maxvonhippel** [comment]
Awesome, thanks @cowlicks!  Let me know if you'd like me to take care of some more tedious aspect or test some list of possible solutions.  I'm studying for an advanced symbolic logic midterm and need things to distract me when I get too wrapped up in circles thinking over the same problems lol.

**Comment by cowlicks** [comment]
@Hainish I poked the chrome devs on [your issue](https://bugs.chromium.org/p/chromium/issues/detail?id=643814&can=2&start=0&num=100&q=&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified&groupby=&sort=).

@maxvonhippel that bug has some more suggestions.

**Comment by maxvonhippel** [comment]
My understanding is that this problem would only really occur for people who are downloading and compiling the source code from GitHub - which (as far as I am aware) would only really be developers and relatively technically savvy people.  In the conversation you ( @cowlicks ) linked to, user `a sargent` wrote:

````
b) Have users download the source themselves, enable developer mode on chrome://extensions, and use the "Load unpacked extension" button load it in developer mode. This is unappealing for various reasons: the steps are arcane for non-developer users, users doing this won't get future autoupdates, and also we issue an annoying warning about developer-mode on every startup (due to malware using it as a vector for force-installing unwanted extensions). 
````

I tried this a week or so ago, and it didn't work for me.  I'm in [developer mode](http://superuser.com/a/768154) but the option to enable privacy badger is still grey-ed out, and the CRX file is grey-ed out when I try the "unpack extension" option.  I am on Version 58.0.3016.0 canary (64-bit), and it shows the blue checkmark saying it's "up to date".

**HOWEVER**, [this](http://superuser.com/a/810394) solution does work for me, and I think it's a great one!  Basically, Google offers a tool which they make and publish on the Google Chrome Web Store for free for running plugins.  This tool actually lets you run whatever you want to run, and it took me all of about five seconds to use it to get Privacy Badger working again on my machine.  

While I agree that it is crucial that the version listed on the Chrome Webstore always work "out of the box", and don't think it's really unreasonable at all to say in the README that if the tool exhibits this bug you can circumvent that by using a tool which Google officially, canonically releases basically to address that specific problem.

Do you (good denizens of this project) agree?  Should I go ahead and submit a PR with a simple set of instructions in the README and the docs for addressing this issue when it comes up?

**Comment by cowlicks** [comment]
@maxvonhippel when you say "this problem would only really occur ..." are you referencing the problem in the chrome issue I linked? That problem is probably related to this. Basically the EFF distributes privacy badger in a special way, and chrome keeps breaking this distribution process. 

The method described in `b)` should work. I think it will add another privacy badger entry in you extensions list, maybe you were looking at the wrong one?

I personally haven't used the solution you linked to, so I'm not sure I can recommend it. It might be good to have a page recommending several options.

**Comment by maxvonhippel** [comment]
@cowlicks you're right on both fronts, I'd misunderstood the unpacked extension load process.  That is definitely a better solution than the google dev plugin solution I linked to.

How about if in the README it says:

> In the event that Chrome does not allow you to enable PrivacyBadger and warns you that PrivacyBadger "is not listed in the Chrome Web Store and may have been added without your knowledge", see *this* troubleshooting page for workarounds.

The *this* in that sentence would then link to a page in the `/docs`, which would have simple instructions with screenshots first for the method listed in `b)` and then if that doesn't work for the alternative method using the google dev plugin.

I was wrong about both `b)` not working and the problem only really occurring for developers.  Sorry about that - I hope my confusion did not lead to any significant confusion for others :)

**Comment by cowlicks** [comment]
@maxvonhippel no worries, confusion is a natural part of the development process.

**Comment by ghostwords** [comment]
This should be fixed as of [Privacy Badger version 2018.1.25](https://github.com/EFForg/privacybadger/releases/tag/release-2018.1.25).

Let me know if updating fixes your problem. You could try forcing an update by visiting `chrome://extensions` and clicking the "Update extensions now" button.
